### PR TITLE
Added additional details to the Developer Tools.

### PR DESCRIPTION
**Pull request by skalavala**
More documentation to help clarify how the Developer Tools is and  how it should be used.

**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

## Checklist:

- [ ] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [ ] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_docs/tools/dev-tools.markdown [hunk 1]
```diff
@@ -24,3 +24,79 @@ Screenshot of Home Assistant's Developer Tools.
 | Templates | <img src='/images/screenshots/developer-tool-templates-icon.png' alt='service developer tool icon' class="no-shadow" height="38" /> | Renders templates |
 | Info | <img src='/images/screenshots/developer-tool-about-icon.png' alt='service developer tool icon' class="no-shadow" height="38" /> | Details about Home Assistant |
 
+## What can I do with Developer Tools?
+The Developer Tools is meant for **all** (not just for the developers) to quickly try out things - like calling services, update states, raising events, and publish messages in mqtt…etc.). It is also a necessary tool for those who write custom automations and scripts by hand. The following describes each of the section in detail.
```
Comment (arsaboo) [inline source/_docs/tools/dev-tools.markdown]: `and publish messages in MQTT, etc.).`

File: source/_docs/tools/dev-tools.markdown [hunk 2]
```diff
@@ -24,3 +24,79 @@ Screenshot of Home Assistant's Developer Tools.
...
+
+{% linkable_title Services %}
+
+This section is used to call Services that are available in the ServiceRegistry.
```
Comment (arsaboo) [inline source/_docs/tools/dev-tools.markdown]: Is it `ServiceRegistry` or `Service Registry`?

File: source/_docs/tools/dev-tools.markdown [hunk 3]
```diff
@@ -24,3 +24,79 @@ Screenshot of Home Assistant's Developer Tools.
...
+
+The list of services in the “Service” drop down are automatically populated based on the components that are found in the configuration, automation and script files.  If a desired service does not exist, it means either the component is not configured properly or not defined in the configuration, automation or script files.
```
Comment (arsaboo) [inline source/_docs/tools/dev-tools.markdown]: We can simplify this

The list of services in the “Service” drop down are automatically populated based on the components that you have added to Home Assistant.

File: source/_docs/tools/dev-tools.markdown [hunk 4]
```diff
@@ -24,3 +24,79 @@ Screenshot of Home Assistant's Developer Tools.
...
+
+Note that this is the state representation of a device within Home Assistant. That means, it is what Home Assistant sees, and it does not communicate with the actual device in any manner. The updated information can still be used to trigger events, and state changes. To communicate with the actual device, it is recommended to call services in the services section above, instead of updating state.
+
+For ex: Changing the `light.bedroom` state from `off` to `on` does not turn on the light. If there is an automation that triggers on the `state` change of the `light.bedroom`, it will be triggered – even though the actual bulb has not turned on. Also, when the bulb state changes – the state information will be overridden. In other words, the changes that are made through the “States” section are temporary, and is recommended to use for testing purposes only.
```
Comment (arsaboo) [inline source/_docs/tools/dev-tools.markdown]: For example, changing...

Comment (arsaboo) [reply source/_docs/tools/dev-tools.markdown]: last line
`...and is recommended for testing purposes only.`

File: source/_docs/tools/dev-tools.markdown [hunk 5]
```diff
@@ -24,3 +24,79 @@ Screenshot of Home Assistant's Developer Tools.
...
+
+{% linkable_title Events %}
+
+This Events section is as basic as it can get. It does only one thing – fires events on the event bus.
```
Comment (arsaboo) [inline source/_docs/tools/dev-tools.markdown]: Remove `This` in the beginning

File: source/_docs/tools/dev-tools.markdown [hunk 6]
```diff
@@ -24,3 +24,79 @@ Screenshot of Home Assistant's Developer Tools.
...
+
+It is a good practice to test the template code in the template editor prior to putting it in automations and scripts.
+
+For more information about jinja2, visit [jinja2 documentation](http://jinja.pocoo.org/docs/dev/templates/), and also read templating document [here](https://home-assistant.io/topics/templating/)
+
+
+{% linkable_title mqtt %}
+
+This section is only visible if the MQTT is configured. To configure MQTT, add `mqtt:` to the `configuration.yaml` file. For more information, refer to [mqtt]( https://home-assistant.io/components/mqtt/)
+Even though MQTT in general provides deeper functionality, the developer tools section of MQTT is limited to publishing messages to a given topic. It supports templates for the payload. To publish a message, simply specify the topic name and the payload and click “PUBLISH” button.
```
Comment (arsaboo) [inline source/_docs/tools/dev-tools.markdown]: Simplify...

`Although MQTT provides greater functionality...`

File: source/_docs/tools/dev-tools.markdown (hunk 3 above)
Comment (skalavala) [reply source/_docs/tools/dev-tools.markdown]: The documentation guidelines says we should not use the words like "you" when writing.  We should change that to something like "The list of services in the “Service” drop down are automatically populated based on the components that are added to Home Assistant.". 

I prefer the statement I had earlier - even though it is a bit longer, it actually makes sense.

Comment (frenck) [reply source/_docs/tools/dev-tools.markdown]: - `drop down` -> `dropdown`
- `a desired` -> `the desired`

File: source/_docs/tools/dev-tools.markdown [hunk 7]
```diff
@@ -24,3 +24,79 @@ Screenshot of Home Assistant's Developer Tools.
...
+## What can I do with Developer Tools?
+The Developer Tools is meant for **all** (not just for the developers) to quickly try out things - like calling services, update states, raising events, and publish messages in mqtt…etc.). It is also a necessary tool for those who write custom automations and scripts by hand. The following describes each of the section in detail.
+
+{% linkable_title Services %}
+
+This section is used to call Services that are available in the ServiceRegistry.
+
+The list of services in the “Service” drop down are automatically populated based on the components that are found in the configuration, automation and script files.  If a desired service does not exist, it means either the component is not configured properly or not defined in the configuration, automation or script files.
+
+When a Service is selected, and if that service requires an `entity_id` to be passed, the “Entity” drop down will automatically be populated with corresponding entities.
```
Comment (frenck) [inline source/_docs/tools/dev-tools.markdown]: - `drop down` -> `dropdown`

File: source/_docs/tools/dev-tools.markdown [hunk 8]
```diff
@@ -24,3 +24,79 @@ Screenshot of Home Assistant's Developer Tools.
...
+
+A Service may also require additional input to be passed. It is commonly referred to as “service data”. The service data is only accepted in the JSON format, and it may be optional depending on the service.
+
+When an entity is selected from the Entity drop down, it automatically populates service data with the corresponding `entity_id`. The service data JSON can then be modified to pass additional \[optional\] parameters. The following is an illustration on how to call a `light.turn_on` service.
+
+To turn on a light bulb, use the following steps:
+1.	Select `light.turn_on` from the Service drop down
+2.	Select the entity (typically the light bulb) from the Entity drop down (if no entity_id is selected, it turns on ALL lights)
```
Comment (frenck) [inline source/_docs/tools/dev-tools.markdown]: - `drop down` -> `dropdown`

File: source/_docs/tools/dev-tools.markdown (hunk 2 above)
Comment (skalavala) [reply source/_docs/tools/dev-tools.markdown]: I wanted to use `ServiceRegistry` as one word, as it also indicates the `ServiceRegistry` component. It resonates well with "real" developers, as well as with non-developer community.

File: source/_docs/tools/dev-tools.markdown (hunk 1 above)
Comment (skalavala) [reply source/_docs/tools/dev-tools.markdown]: I'm fine with that.

File: source/_docs/tools/dev-tools.markdown [hunk 9]
```diff
@@ -24,3 +24,79 @@ Screenshot of Home Assistant's Developer Tools.
...
+The Template Editor provides a way to test the template code quickly. When the Template Editor page is loaded, it comes with a sample template code that illustrates how the code can be written and tested. 
+
+It has two sections, code goes on the left hand side, and the output is shown on the right hand side. The code can be removed and replaced, and when the page is loaded/refreshed, the default sample code will be loaded back. 
+
+It is a good practice to test the template code in the template editor prior to putting it in automations and scripts.
+
+For more information about jinja2, visit [jinja2 documentation](http://jinja.pocoo.org/docs/dev/templates/), and also read templating document [here](https://home-assistant.io/topics/templating/)
+
+
+{% linkable_title mqtt %}
```
Comment (frenck) [inline source/_docs/tools/dev-tools.markdown]: - `mqtt` -> `MQTT`

File: source/_docs/tools/dev-tools.markdown [hunk 10]
```diff
@@ -24,3 +24,79 @@ Screenshot of Home Assistant's Developer Tools.
...
+
+This section is only visible if the MQTT is configured. To configure MQTT, add `mqtt:` to the `configuration.yaml` file. For more information, refer to [mqtt]( https://home-assistant.io/components/mqtt/)
+Even though MQTT in general provides deeper functionality, the developer tools section of MQTT is limited to publishing messages to a given topic. It supports templates for the payload. To publish a message, simply specify the topic name and the payload and click “PUBLISH” button. 
+
+{% linkable_title Info %}
+
+The Information tab simply provides information about the current installed version, additional links and credits. The tab also contains a section that shows `syslog` information, and the contents of `home-assistant.log` with an option to clear and refresh the logs.
```
Comment (frenck) [inline source/_docs/tools/dev-tools.markdown]: - `current` -> `currently`
- `information, and the contents` -> `information and the contents`

File: source/_docs/tools/dev-tools.markdown (hunk 5 above)
Comment (skalavala) [reply source/_docs/tools/dev-tools.markdown]: I like it the way it is.

File: source/_docs/tools/dev-tools.markdown (hunk 6 above)
Comment (skalavala) [reply source/_docs/tools/dev-tools.markdown]: I am good with this proposal. There are many ways of saying, it, whatever makes sense to everyone!

File: source/_docs/tools/dev-tools.markdown (hunk 4 above)
Comment (skalavala) [reply source/_docs/tools/dev-tools.markdown]: I think we are sending too much time going back and forth. It is crazy! Grammatically, it was correct.

File: source/_docs/tools/dev-tools.markdown (hunk 7 above)
Comment (skalavala) [reply source/_docs/tools/dev-tools.markdown]: I'm good with `dropdown` as one word. `A service` is contextually correct compared to `The service`.

File: source/_docs/tools/dev-tools.markdown (hunk 9 above)
Comment (skalavala) [reply source/_docs/tools/dev-tools.markdown]: Good!

File: source/_docs/tools/dev-tools.markdown (hunk 10 above)
Comment (skalavala) [reply source/_docs/tools/dev-tools.markdown]: yep!, good one!

**Comment by frenck** [comment]
Sorry @skalavala @cogneato, Why is this merged? Two reviewers requested changes.
Why not updating this PR?

**Comment by skalavala** [comment]
@frenck I will be doing another PR right away, as I also added additional required images for this. I will be making some changes in my next PR. I just don't know how to do all the merging in place.

**Pull request merged**